src/main/java/hudson/plugins/jira/selector/perforce/PerforceJobIssueSelector.java

```diff
@@ -0,0 +1,78 @@
+package hudson.plugins.jira.selector.perforce;
+
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import com.tek42.perforce.model.Changelist;
+import com.tek42.perforce.model.Changelist.JobEntry;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.jira.JiraSite;
+import hudson.plugins.jira.Messages;
+import hudson.plugins.jira.RunScmChangeExtractor;
+import hudson.plugins.jira.selector.AbstractIssueSelector;
+import hudson.plugins.perforce.PerforceChangeLogEntry;
+import hudson.scm.ChangeLogSet;
+import hudson.scm.ChangeLogSet.Entry;
+
+@Extension(optional = true)
```

Comment:
I am not sure what you are referring to. These are two different classes. Each of them is marked as an extension. Feel free to modify that code if you do not like it. 
I do not have any more time to test if @Extension annotates nested classes automatically and documenation does not state it clearly.